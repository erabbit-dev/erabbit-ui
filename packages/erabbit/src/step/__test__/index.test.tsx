import { mount } from '@vue/test-utils'
import { later } from '../../test'
import { Step, StepItem } from '..'
import { ref } from 'vue'
import { expect } from 'vitest'

test('should render default Step', () => {
  const wrapper = mount(
    <Step>
      <StepItem></StepItem>
      <StepItem></StepItem>
      <StepItem></StepItem>
    </Step>,
  )
  expect(wrapper.html()).toMatchSnapshot()
})
test('should render  Step with title and desc', () => {
  const wrapper = mount(
    <Step>
      <StepItem title="开始" desc="2023-09-28 10:23:45"></StepItem>
      <StepItem title="中间" desc="2023-09-28 10:25:46"></StepItem>
      <StepItem title="结束" desc="2023-09-28 10:26:03"></StepItem>
    </Step>,
  )
  expect(wrapper.html()).toMatchSnapshot()
})
test('should render Step with active first one item', () => {
  const wrapper = mount(
    <Step active-index={1}>
      <StepItem></StepItem>
      <StepItem></StepItem>
      <StepItem></StepItem>
    </Step>,
  )
  expect(wrapper.html()).toMatchSnapshot()
})
test('should render  Step with different size', () => {
  const wrapper = mount({
    setup() {
      return () => (
        <div>
          <Step active-index={1} size="medium">
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
          </Step>
          <Step active-index={1} size="mini">
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
          </Step>
          <Step active-index={1} size="small">
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
          </Step>
        </div>
      )
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
})
test('should render Step with vertical mode', () => {
  const wrapper = mount(
    <Step active-index={1} mode="vertical">
      <StepItem></StepItem>
      <StepItem></StepItem>
      <StepItem></StepItem>
    </Step>,
  )
  expect(wrapper.html()).toMatchSnapshot()
})
test('should change acitve item by btn click trigger', async () => {
  const activeIndex = ref(-1)
  const stepRef = ref(null)
  const wrapper = mount({
    setup() {
      return () => (
        <div>
          <button onClick={() => activeIndex.value++}></button>
          <Step ref={stepRef} active-index={activeIndex.value} mode="vertical">
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
            <StepItem></StepItem>
          </Step>
        </div>
      )
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.find('button').trigger('click')
  await later(100)
  expect(stepRef.value.getActiveIndex()).toBe(0)
  wrapper.find('button').trigger('click')
  await later(100)
  expect(stepRef.value.getActiveIndex()).toBe(1)
  wrapper.find('button').trigger('click')
  await later(100)
  expect(stepRef.value.getActiveIndex()).toBe(2)
  wrapper.find('button').trigger('click')
  await later(100)
  expect(stepRef.value.getActiveIndex()).toBe(3)
  wrapper.find('button').trigger('click')
  await later(100)
  expect(stepRef.value.getActiveIndex()).toBe(4)
  wrapper.find('button').trigger('click')
  await later(100)
  expect(stepRef.value.getActiveIndex()).toBe(5)
  wrapper.find('button').trigger('click')
  await later(100)
  expect(stepRef.value.getActiveIndex()).toBe(6)
})

test('should add or remove a step by button event', async () => {
  const wrapper = mount({
    setup() {
      const list = ref<{
        title: string
        desc: string
      }>([])
      const addStep = () => {
        list.value.push({
          title: `标题${~~(Math.random() * 100)}`,
          desc: `副标题${~~(Math.random() * 100)}`,
        })
      }
      const removeStep = (index) => {
        list.value.splice(index, 1)
      }
      return () => (
        <div>
          <button class="addStep" onClick={addStep}>
            添加步骤
          </button>
          <button
            class="removeStep"
            onClick={() => removeStep(list.value.length - 1)}
          >
            移除步骤
          </button>
          <Step>
            {list.value.map((item, index) => (
              <StepItem
                key={index}
                title={item.title}
                desc={item.desc}
              ></StepItem>
            ))}
          </Step>
        </div>
      )
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
  await later()
  wrapper.find('.addStep').trigger('click')
  await later()
  expect(wrapper.findAll('.er-steps-item').length).toBe(1)
  wrapper.find('.addStep').trigger('click')
  await later()
  expect(wrapper.findAll('.er-steps-item').length).toBe(2)
  wrapper.find('.addStep').trigger('click')
  await later()
  expect(wrapper.findAll('.er-steps-item').length).toBe(3)
  wrapper.find('.removeStep').trigger('click')
  await later()
  expect(wrapper.findAll('.er-steps-item').length).toBe(2)
  wrapper.find('.removeStep').trigger('click')
  await later()
  expect(wrapper.findAll('.er-steps-item').length).toBe(1)
  wrapper.find('.removeStep').trigger('click')
  await later()
  expect(wrapper.findAll('.er-steps-item').length).toBe(0)
})
