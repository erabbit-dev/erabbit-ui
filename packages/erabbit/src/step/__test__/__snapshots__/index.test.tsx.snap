// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should add or remove a step by button event 1`] = `
"<div><button class=\\"addStep\\">添加步骤</button><button class=\\"removeStep\\">移除步骤</button>
  <div class=\\"er-steps default horizontal\\"></div>
</div>"
`;

exports[`should change acitve item by btn click trigger 1`] = `
"<div><button></button>
  <div class=\\"er-steps default vertical\\">
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"vertical-main\\">
        <div class=\\"title\\"></div>
        <div class=\\"desc\\"></div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`should render  Step with different size 1`] = `
"<div>
  <div class=\\"er-steps medium horizontal\\">
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
  </div>
  <div class=\\"er-steps mini horizontal\\">
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
  </div>
  <div class=\\"er-steps small horizontal\\">
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
    <div class=\\"er-steps-item active\\">
      <div class=\\"step\\"><span>0</span></div>
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
  </div>
</div>"
`;

exports[`should render  Step with title and desc 1`] = `
"<div class=\\"er-steps default horizontal\\">
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\">开始</div>
    <div class=\\"desc\\">2023-09-28 10:23:45</div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\">中间</div>
    <div class=\\"desc\\">2023-09-28 10:25:46</div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\">结束</div>
    <div class=\\"desc\\">2023-09-28 10:26:03</div>
  </div>
</div>"
`;

exports[`should render Step with active first one item 1`] = `
"<div class=\\"er-steps default horizontal\\">
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\"></div>
    <div class=\\"desc\\"></div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\"></div>
    <div class=\\"desc\\"></div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\"></div>
    <div class=\\"desc\\"></div>
  </div>
</div>"
`;

exports[`should render Step with vertical mode 1`] = `
"<div class=\\"er-steps default vertical\\">
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"vertical-main\\">
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"vertical-main\\">
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"vertical-main\\">
      <div class=\\"title\\"></div>
      <div class=\\"desc\\"></div>
    </div>
  </div>
</div>"
`;

exports[`should render default Step 1`] = `
"<div class=\\"er-steps default horizontal\\">
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\"></div>
    <div class=\\"desc\\"></div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\"></div>
    <div class=\\"desc\\"></div>
  </div>
  <div class=\\"er-steps-item active\\">
    <div class=\\"step\\"><span>0</span></div>
    <div class=\\"title\\"></div>
    <div class=\\"desc\\"></div>
  </div>
</div>"
`;
